src/main.rs

```diff
@@ -234,25 +228,37 @@ fn run() -> i32 {
         }
     };
 
-    if path == "-" {
-        let mut buf = Vec::new();
-        image
-            .write_to(&mut io::Cursor::new(&mut buf), output_format)
-            .expect("Encoding failed");
-        io::stdout()
-            .write_all(buf.as_slice())
-            .expect("Writing to stdout failed");
+    let writer = if path == "-" {
+        either::Left(io::stdout())
     } else {
         match File::create(&Path::new(&path)) {
-            Ok(mut f) => image
-                .write_to(&mut f, output_format)
-                .expect("Writing to file failed"),
+            Ok(f) => either::Right(f),
             Err(e) => {
                 eprintln!("Failed to create {}: {}", path, e);
                 return 1;
             }
         }
-    }
+    };
+
+    let encoder = match output_ext.as_ref() {
+        "png" => either::Left(image::codecs::png::PngEncoder::new(writer)),
+        "pam" => either::Right(
+            image::codecs::pnm::PnmEncoder::new(writer)
+                .with_subtype(image::codecs::pnm::PnmSubtype::ArbitraryMap),
+        ),
+        _ => {
+            eprintln!("Invalid image format specified");
```

Comment:
This will leave an empty file where none would be created previously. It's probably a good idea to bail early as well so we don't actually do any of the screenshotting work user input is invalid.